clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1842,5 +1842,6 @@ def err_hlsl_unexpected_end_of_params
 def err_hlsl_rootsig_repeat_param : Error<"specified the same parameter '%0' multiple times">;
 def err_hlsl_rootsig_missing_param : Error<"did not specify mandatory parameter '%0'">;
 def err_hlsl_number_literal_overflow : Error<"integer literal is too large to be represented as a 32-bit %select{signed |}0 integer type">;
+def err_hlsl_rootsig_non_zero_flag : Error<"non-zero integer literal specified for flag value">;
```

Comment:
If we're going to use a specific error message here it would be clearer for the user to say what we need here. Something like "flag value is neither a literal 0 nor a named value" maybe.